### ISSUE TITLE
wrong hidden layer dim in denoising autoencoder

### ISSUE THREAD

**Issue by huikang** [PacktPublishing/Deep-Learning-with-TensorFlow-2-and-Keras#3]
Based on the definition of denoising auto encoder, the hidden layer's dim should > input layer,

is this a bug in the 2nd cell at

https://github.com/PacktPublishing/Deep-Learning-with-TensorFlow-2-and-Keras/blob/master/Chapter%209/DenoisingAutoencoder.ipynb

**Comment by sujitpal** [comment]
Can you please provide a link to the definition you referenced above?

In general, an autoencoder works by reducing the dimensionality of the input and then expanding it back again. The intuition is that only useful features are retained in the compression, so the autoencoder gets trained to extract a compressed representation of the original. Based on that the hidden dimension should be smaller than the original dimension.

Example uses MNIST images which are (28, 28, 1) images, they are flattened to (784,). Encoder compresses it to a hidden dimension of (128,) and then decoder expands it again to (784,).
